pkg/fpga/bitstream/aocx.go

```diff
@@ -172,11 +168,6 @@ func parseFpgaBin(d []byte) (*FileGBS, error) {
 		return nil, err
 	}
 
-	if afuUUID := g.AcceleratorTypeUUID(); afuUUID != OpenCLUUID {
-		g.Close()
-		return nil, errors.Errorf("incorrect OpenCL BSP AFU UUID (%s)", afuUUID)
-	}
```

Comment:
I understand the reasoning here, but still a bit concerned about removing validation code. @kad WDYT?

Reply:
As I recall that constant `OpenCLUUID` was a "well known magic ID" for OpenCL based bitstreams. If newer versions of OpenCL SDK have something else, it would be good to add new magic IDs, and do verification for any of them.

Reply:
> I understand the reasoning here, but still a bit concerned about removing validation code. @kad WDYT?

I don't think that kind of validation should be part of the parser. AOCXs should be treated as common ELF binaries with specific sections. You can always run such a check afterwards.

> As I recall that constant `OpenCLUUID` was a "well known magic ID" for OpenCL based bitstreams. If newer versions of OpenCL SDK have something else, it would be good to add new magic IDs, and do verification for any of them.

It seems that even the same OpenCL BSP may use different AFU UUIDs for different board variants: https://github.com/intel/intel-device-plugins-for-kubernetes/issues/1642#issuecomment-1878824101

Reply:
If the bitstream's UUID can change and still be valid, plugin would always be "one step behind". At least we should allow admin to provide an UUID that is accepted without changing the plugin code?

UUID appears to just be a field in the binary's metadata. Does it serve any security purpose?

Reply:
originally that UUID was used as part of validation that AOCX is containing something expected, to prevent malicious content. It was also original idea about to distinguish "normal" bitstreams programmed to FPGA (which can be re-used once programmed) from OpenCL ones (which we can't trust as identifier, as the interfaceID + afuID pair is not unique and AOCX bitstream should be programmed every time).
Theoretically it would be good to consult FPGA OpenCL BSP people to validate what they are planning to use.
Temporarily, I'm fine if this check will be commented out (not deleted) with TODO: label to verify in future.

Reply:
`aocl binedit` which is also an AOCX parser doesn't run such a check. `aocl program` either.

Reply:
as I mentioned above, I'm ok for commenting out this check until we will get more problems.

Reply:
@eliaskoromilas can you amend the PR as @kad suggested?

Reply:
I tested with more OpenCL BSPs (7 board variants in total) and these two UUIDs are being repeated. Would you be more comfortable with using a (mutable) slice and keeping the check?

The slice would be exposed and anyone who uses the library (including myself) would be able to populate it with more, if needed.

Reply:
Thanks for effort of testing @eliaskoromilas!
Commenting out or check via slice of known UUIDs are both fine for me.
For now, I'd probably go with commenting check out, as it is smaller effort. Later check with slice of known OpenCL UUID can be re-added as separate PR.